### PR TITLE
Adding github action for link checking

### PR DESCRIPTION
**Pull request by deining**
This PR adds a html lint action to the GitHub workflow.
I leave fixing of reported errors as an exercise to you 😁.

### CODE REVIEW

File: .github/workflows/htmltest.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+jobs:
+  htmltest:
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Fetch all history for .GitInfo and .Lastmod
```
Comment (diemol) [inline .github/workflows/htmltest.yml]: I don't think we need to check the whole history to check links.

**Review by diemol — CHANGES_REQUESTED** [review]
Thank you, @deining! Left a minor comment.

File: .github/workflows/htmltest.yml (hunk 1 above)
Comment (deining) [reply .github/workflows/htmltest.yml]: > I don't think we need to check the whole history to check links.

That's very true. I removed the with clause, this certainly will make checkout faster.
You may have to adapt the link checker's configuration file according to your needs, too.

File: .github/workflows/htmltest.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: "latest"
```
Comment (diemol) [inline .github/workflows/htmltest.yml]: I missed this. I prefer to pin the Hugo version and upgrade when we have time to check on things. What is your experience with this?

Comment (deining) [reply .github/workflows/htmltest.yml]: > I missed this. I prefer to pin the Hugo version and upgrade when we have time to check on things. What is your experience with this?

Yes, that is of course the better setting in your scenario. Changed now.

**Review by diemol — APPROVED** [review]
Thank you, @deining!

**Comment by diemol** [comment]
It does seem the job is failing and I cannot see why. Do you have any successful execution to see how the results look like? @deining

**Comment by diemol** [comment]
It also seems to take 1.5 hours to run? That is a bit too much when pushing commits and opening PRs

**Comment by deining** [comment]
> It does seem the job is failing and I cannot see why. Do you have any successful execution to see how the results look like? @deining

Have a look here:

https://github.com/deining/docsy/actions/runs/4281770427

**Comment by deining** [comment]
> It also seems to take 1.5 hours to run? That is a bit too much when pushing commits and opening PRs

Definitely!

I see several possibilities to bring run time down:

```
We store temporary files in tmp/.htmltest by default.
This contains a log of output and a cache of external links,
speeding up subsequent runs.
```

There is also a `-s` switch:

```
  -s, --skip-external          Skip external link checks, may shorten execution
                               time considerably.
```

I would start by running [`htmltest`](https://github.com/wjdp/htmltest/releases) locally and fix the reported issues. Once alle issues ae fixed, I would run the `htmltest` workflow again and is run time is acceptable for you.

I also experimented with [hyperlink](https://github.com/untitaker/hyperlink) which is blazingly fast, but perform internal link checks only. If you want to give it a try, have a look at this [wokflow file](https://github.com/deining/docsy-example/blob/deploy-to-github-pages/.github/workflows/link-check.yml) I authored.

**Comment by diemol** [comment]
The thing is that I am really short of time, we are only a few working on this project and adding this would increase the workload because we need to have this working well so it does not block PRs. Seems I need to leave this open and revisit whenever we have time. Sorry. 🥲

**Comment by deining** [comment]
I switched the engine, the workflow now uses [hyperlink](https://github.com/untitaker/hyperlink) as link checker. Link check lasts 5 sec only, only internal links are checked.

**Review by diemol — APPROVED** [review]
Thank you, @deining!

I think, at least for now, we care more about our internal links since we do not use extensively external links.

**Pull request merged**